arch/arm64/boot/dts/qcom/sm7150-xiaomi-davinci.dts

```diff
@@ -151,7 +151,44 @@
 	clock-frequency = <100000>;
 	status = "okay";
 
-	/* goodix,gt9889 (touchscreen) @ 5d */
+	goodix@5d {
+		compatible = "goodix,gt9889";
+		reg = <0x5d>;
+		status = "okay";
+		interrupt-parent = <&tlmm>;
+		interrupts = <9 0x2800>;
+		vtouch-supply = <&vreg_l6c_3p0>;//3.3v
+		vtouch-load = <600000>;
+		pinctrl-names = "pmx_ts_active","pmx_ts_suspend";
```

Comment:
Let's use standardized default and sleep:
pinctrl-names = "default", "sleep";

This requires changes to the driver. (you previously tested this and got errors because the driver expects  pmx_ts_active and pmx_ts_suspend names)